common/src/gadgets/cond_add.rs

```diff
@@ -3,32 +3,70 @@ use ark_ff::{FftField, Field};
 
 use crate::domain::Domain;
 use crate::gadgets::booleanity::BitColumn;
-use crate::AffineColumn;
+use crate::{AffineColumn, FieldColumn};
 
-pub trait CondAdd<F, AffinePoint>
+use super::{ProverGadget, VerifierGadget};
+
+/// Affine point with conditional add implementation.
+///
+/// Currently supported for Arkworks Short Weierstrass and Twisted Edwards affine points.
+pub trait AffineCondAdd: AffineRepr
```

Comment:
I actually had my tests failing because I set the point to TE but the CondAdd of SW when I was calling the verifier :-/ This design prevents that.